In [1]:
import pandas as pd

from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error

from joblib import dump

In [2]:
df = pd.read_csv('train_data.csv')

In [3]:
features = df.drop(['last_price'], axis=1)
target = df['last_price'] / 1000000

In [4]:
# criterion='mse' by default
model = RandomForestRegressor(random_state=12345, n_estimators=3)

In [5]:
model.fit(features, target)

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
                      max_features='auto', max_leaf_nodes=None,
                      min_impurity_decrease=0.0, min_impurity_split=None,
                      min_samples_leaf=1, min_samples_split=2,
                      min_weight_fraction_leaf=0.0, n_estimators=3, n_jobs=None,
                      oob_score=False, random_state=12345, verbose=0,
                      warm_start=False)

In [6]:
predictions = model.predict(features)

In [7]:
mse = mean_squared_error(target, predictions)
rmse = mse**0.5

print("RMSE Обучающая выборка:", rmse)

RMSE Обучающая выборка: 4.229147624009398


In [7]:
dump(model, 'random_forest_regressor.joblib')

['random_forest_regressor.joblib']

RMSE:

* Обучающая выборка: 4.229147624009398
* Тестовая выборка: 8.8120811642178

Лучше дерева может быть только лес: RMSE на тестовой выборке ниже! На обучающей — это значение не так важно из-за переобучения.